src/mesh/rbf.f90

```diff
@@ -0,0 +1,597 @@
+! LOCAL VARIABLES
+REAL             :: xi
+REAL,PARAMETER   :: a=10E-3
+!===================================================================================================================================
+! normalized distance
+xi = dist/supportRadius
+
+SELECT CASE(RBFType)
+! Local support RBFs
+CASE(1)
+  IF (xi.GT.1.) THEN
+    EvaluateRBF = 0.
+  ELSE
+    EvaluateRBF = (1.-xi)**2.
+  END IF
+CASE(2)
+  IF (xi.GT.1.) THEN
+    EvaluateRBF = 0.
+  ELSE
+    EvaluateRBF=(1.-xi)**4.*(4.*xi+1.)
+  END IF
+CASE(3)
+  IF (xi.GT.1.) THEN
+    EvaluateRBF = 0.
+  ELSE
+    EvaluateRBF = (1.-xi)**6.*((35./3.)*(xi**2.)+6.*xi+1.)
+  END IF
+CASE(4)
+  IF (xi.GT.1.) THEN
+    EvaluateRBF = 0.
+  ELSE
+    EvaluateRBF = (1.-xi)**8.*(32.*(xi**3.)+25.*(xi**2.)+8.*xi+1.)
+  END IF
+CASE(5)
+  IF (xi.GT.1.) THEN
+    EvaluateRBF = 0.
+  ELSE
+    EvaluateRBF = (1.-xi)**5.
+  END IF
+CASE(6)
+  IF (xi.GT.1.) THEN
+    EvaluateRBF = 0.
+  ELSE
+    EvaluateRBF = 1.+(80./3.)*(xi**2.)-40.*(xi**3.)+15.*(xi**4.)-(8./3.)*(xi**5.)+20.*(xi**2.)*LOG(xi)
+  END IF
+  IF (xi.LT.10E-13) EvaluateRBF=1.
+CASE(7)
+  IF (xi.GT.1.) THEN
+    EvaluateRBF = 0.
+  ELSE
+    EvaluateRBF = 1.-30.*(xi**2.)-10.*(xi**3.)+45.*(xi**4.)-6.*(xi**5.)-60.*(xi**3.)*LOG(xi)
+  END IF
+  IF (xi.LT.10E-13) EvaluateRBF=1.
+CASE(8)
+  IF (xi.GT.1.) THEN
+    EvaluateRBF = 0.
+  ELSE
+    EvaluateRBF = 1.-20.*(xi**2.)+80.*(xi**3.)-45.*(xi**4.)-16.*(xi**5.)+60.*(xi**4.)*LOG(xi)
+  END IF
+  IF (xi.LT.10E-13) EvaluateRBF=1.
+! Global support RBFs
+CASE(9)
+  EvaluateRBF = (xi**2.)*LOG(xi)
+  IF (xi.LT.10E-13) EvaluateRBF=0.
+CASE(10)
+  EvaluateRBF = SQRT((a**2.)+(xi**2.))
+CASE(11)
+  EvaluateRBF = SQRT(1./((a**2.)+(xi**2.)))
+CASE(12)
+  EvaluateRBF = 1.+(xi**2.)
+CASE(13)
+  EvaluateRBF = 1./(1.+(xi**2.))
+CASE(14)
+  EvaluateRBF = EXP(-(xi**2.))
+CASE DEFAULT
+  CALL Abort(__STAMP__,'RBF Type is unknown')
+END SELECT
+
+END FUNCTION EvaluateRBF
+
+
+FUNCTION Distance(x,y)
+!===================================================================================================================================
+! Calculate euclidean distance between two points.
+!===================================================================================================================================
+! INPUT VARIABLES
+REAL,INTENT(IN) :: x(1:3)    ! Coordinates of first point
+REAL,INTENT(IN) :: y(1:3)    ! Coordinates of second point
+!-----------------------------------------------------------------------------------------------------------------------------------
+! OUTPUT VARIABLES
+REAL            :: Distance
+!===================================================================================================================================
+
+! d(x,y) = \|x-y\|_2 = \sqrt{(x_{1} - y_{1})^2 + \cdots + (x_{n} - y_{n})^2} = \sqrt{\sum_{i=1}^n (x_i-y_i)^2}
+Distance = SQRT((x(1)-y(1))**2.+(x(2)-y(2))**2.+(x(3)-y(3))**2.)
```

Comment:
I would prefer to use `VECNORM(x-y)` at the function calls and eliminate this special function.

Reply:
Replaced by NORM2, which is Fortran 2008 standard and should by now be supported by every relevant compiler. Otherwise replace by (SQRT(DOT_PRODUCT)).